File: annotation-processors/src/main/java/com/comsysto/livingdoc/annotation/processors/plantuml/model/RelationPart.java
package com.comsysto.livingdoc.annotation.processors.plantuml.model;

import com.comsysto.livingdoc.annotation.plantuml.PlantUmlClass;
import com.comsysto.livingdoc.annotation.plantuml.PlantUmlExecutable;
import com.comsysto.livingdoc.annotation.processors.plantuml.PlantUmlClassDiagramProcessor;

@PlantUmlClass(diagramIds = PlantUmlClassDiagramProcessor.DIAGRAM_ID)
public interface RelationPart {

    @PlantUmlExecutable
    RelationId getId();
}

File: annotation-processors-kotlin/src/main/kotlin/model/RelationPart.kt
package com.comsysto.livingdoc.kotlin.annotation.processors.model

import javax.lang.model.element.TypeElement

/**
 * A diagram part that represents a relation between two type elements.
 */
data class RelationPart(val id: RelationId, val left: TypeElement, val right: TypeElement, val relation: Relation)

enum class Relation {

    /**
     * The right type realizes (implements) the left type.
     */
    REALIZATION,

    /**
     * The right type extends the left type.
     */
    INHERITANCE,

    /**
     * The left type references the right type in form of a directed
     * association.
     */
    ASSOCIATION
}
